pkg/runner/writer_http.go

```diff
@@ -0,0 +1,152 @@
+/*
+Copyright 2023 API Testing Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package runner
+
+import (
+	"bytes"
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"mime/multipart"
+	"net/http"
+	"os"
+
+	"github.com/linuxsuren/api-testing/pkg/apispec"
+)
+
+type httpResultWriter struct {
+	requestMethod string
+	targetUrl     string
+	parameters    map[string]string
+	templateFile  string
+}
+
+// NewHTTPResultWriter creates a new httpResultWriter
+func NewHTTPResultWriter(requestType string, url string, parameters map[string]string, templateFile string) ReportResultWriter {
+	return &httpResultWriter{
+		requestMethod: requestType,
+		targetUrl:     url,
+		parameters:    parameters,
+		templateFile:  templateFile,
+	}
+}
+
+// Output writes the JSON base report to target writer
+func (w *httpResultWriter) Output(result []ReportResult) (err error) {
+	url := w.targetUrl
+	for key, value := range w.parameters {
+		if url == w.targetUrl {
+			url = fmt.Sprintf("%s?%s=%s", url, key, value)
+		} else {
+			url = fmt.Sprintf("%s&%s=%s", url, key, value)
+		}
+	}
+	log.Println(url)
+
+	var req *http.Request
+	if w.templateFile == "" {
+		// 使用默认模板将数据序列化为 JSON 格式
```

Comment:
Please write the comment in source files in English instead of Chinese. You can write the comment in PRs or issues in Chinese.

Reply:
ok